index.md

```diff
@@ -652,6 +683,47 @@ const SchemaFormWithPlugin = SchemaFormFactory([
 
 Now our schema will correctly pass the `label` property into our `FormText` component.
 
+Just like `componentProp`, `mapProps` can also receive a function to handle advanced property parsing logic.
+
+If a function is provided, the plugin will run the function before parsing each element to retrieve the mapping of properties. The function will inject the current element as the first parameter of the function.
+
+**Important:** The mapping of props is done **after** the replacement of the `component` base property. Make sure to take this into consideration if you are also replacing a property for `component` using `componentProp`.
+
+Consider the following schema and example.
+
+```json
+{
+  "firstName": {
+    "type": "FormText",
+    "label": "First name",
+    "important": true
+  },
+  "firstName": {
+    "field": "FormText",
+    "label": "First name",
+    "important": true
+  }
+}
+```
+
+```js
+const SchemaFormWithPlugin = SchemaFormFactory([
+  LookupPlugin({
+    mapProps: (el) => {
+      // Map important to required only for 'First name'
+      if (el.label === 'First name') {
+        return {
+          important: 'required'
```

Comment:
So this usage feels is slightly unfamiliar to me, considering `mapSomething` would make me expect something similar to `<Array>.map()`. For example, how to remove certain fields if we don’t want them? 

I know this is coming a bit late given that the pluign is already written but those docs made me think about it some more.
What do you @marina-mosti @tzhelyazkova think about the below API examples?

```js
LookupPlugin({
  field: (val, el) => ({ component: getComponentForField(val) }), // renaming prop and adjusting the value
  important: (val, el) => {
    if (el.label === 'First name') {
      return { required: val === 'yes' ? true : false } // replaces 'important' with 'required' while adapting its value
    } else return {} // makes `important` disappear from the schema. Can also return `false` for the same effect.
  },
  notImportantProp: false, // makes 'notImportantProp' disappear. Alternatively () => ({})
  info: 'label' // supporting basic prop renaming
})
```
Where `val` is the current value of the prop (where you might want to change from a string `"yes"` into `true`).
Where `el` is the whole field object.

Alternatively if passing just a function
```js
LookupPlugin(el => {
  const { field, ...props } = el
  const component = getComponentForField(field)
  return {
    ...props,
    component
  }
})
```

Just exploring ideas here. 😅

Reply:
I like the idea of being able to remove a prop, i'll think about this implementation proposal. The only first problem i see is that using `false` makes it so that the schema itself can not use `false` as a valid prop value, same with returning `{}`

It may be overkill, but maybe were looking at a `deleteProps` object? Not sure yet